tier4/driving_log_replayer
# feat(localization): add component_state_monitor in localization scenario

## Types of PR


## Description
I would like to add a simple evaluation criteria based on the availability of `/localization/pose_estimator/pose_with_covariance`, which is an output of `ndt_scan_matcher`. 

The original localization scenario implementation failed to detect a failure case where NDT fails due to some reason (e.g. node shutdowns due to runtime error) while running, since the "Convergence" and "Reliability" only cares about messages that have been already received. 

Thus, I'd like to add component_state_monitor besides the localization_evaluator so that it diagnoses the availability of ndt outputs.

## How to review this PR
- Run tutorial localization scenario, and confirm that the test passes
- Run tutorial localization scenario, kill `ndt_scan_matcher` with `kill NDT_PID`, and confirm that the test fails

In addition, I'd like to get some advice on:
- if it is OK to fix minor typo in `generate.py`
- if it is safe to change the output summary (self._summary in `LocalizationResult`) format

## Others
